src/main/java/org/jenkinsci/plugins/build_token_root/BuildRootAction.java

```diff
@@ -150,4 +167,57 @@ private void ok(StaplerResponse rsp) throws IOException {
         w.close();
     }
 
+    // --- adapter methods for Job type which do follow AbstractProject semantic but don't inherit AbstractProject.
+
+    private BuildAuthorizationToken getAuthToken(Job<?, ?> p) {
+        if (p instanceof AbstractProject) return ((AbstractProject) p).getAuthToken();
+
+        try {
+            Method m = p.getClass().getMethod("getAuthToken");
+            return (BuildAuthorizationToken) m.invoke(p);
+        } catch (Exception e) {
+            // Job does not support BuildAuthorizationToken
+        }
+        return null;
+    }
+
+    private boolean isDisabled(Job<?, ?> p) {
+        if (p instanceof AbstractProject) return ((AbstractProject) p).isDisabled();
+
+        try {
+            Method m = p.getClass().getMethod("getAuthToken");
```

Comment:
Is there a copy & paste error here? Should it not be trying to get a method named `isDisabled`?